SalesforceCommerceCloud/pwa-kit
# ShopperCustomer hooks and mutations

# Description



The PR implements the ShopperCustomer query hooks and mutations action detailed in the ticket.

There is a Phase 2 where we'll implement the rest of the ShopperCustomer query hooks and mutations actions.


### 📋 TODOs
- [x] Create a Phase 2 ticket to implement the remaining ShopperCustomer query hooks and mutations actions.
- [x] Fine grain invalidation of '/customers' during mutations.
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/53af03c2401b826c0a102ba375f9e1a41a31468b/packages/commerce-sdk-react/src/hooks/ShopperCustomers/mutation.ts#L241
- [ ] Fix mutation test failing in CI but passing locally:
```
npm run test:update-mocked-responses src/hooks/ShopperCustomers/mutation.test.tsx
```
- [ ] Mutation action `getResetPasswordToken` is not working properly. Probably related to issue scoping tokens. https://github.com/SalesforceCommerceCloud/pwa-kit/blob/ff93b1dac33701757c9a94b3474ed6ed8657e8d3/packages/test-commerce-sdk-react/app/pages/use-shopper-customer.tsx#L114-L115
- [ ] Implement the flow: Login as a guest user and then registered that user. Currently Login as a guest is not working.
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/ff93b1dac33701757c9a94b3474ed6ed8657e8d3/packages/test-commerce-sdk-react/app/pages/use-shopper-customer.tsx#L84-L85
- [ ] Create a basket for the register user before testing if the basket exist. Currently Baskets are created in PR not merged yet https://github.com/SalesforceCommerceCloud/pwa-kit/pull/768
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/a0009a74ae60a20e26e78e69108d666fa167f0e4/packages/commerce-sdk-react/src/hooks/ShopperCustomers/query.test.tsx#L121-L122


# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- First iteration implementing ShopperCustomer hooks and mutations.

# How to Test-Drive This PR

- Start the test project and navigate to http://localhost:3000/customer
- Click the login button and verify Query hooks and Mutations work as expected.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
